components/categ-tool-group.vue

```diff
@@ -0,0 +1,261 @@
+<template>
+
+    <b-row>
+        <b-col cols="6">
+
+            <!-- Heading for tool grouping component -->
+            <b-row>
+                <h3>{{ uiText.title }}</h3>
+            </b-row>
+
+            <!-- Instructions prompting the user how to group tools -->
+            <b-row>
+                <p class="instructions-text">
+                    {{ uiText.instructions }}
+                </p>
+            </b-row>
+
+            <!-- Tool grouping inputs -->
+
+            <!-- Tool group name -->
+            <b-row>
+                <b-input-group>
+                    <label for="tool-name-textbox">Group name:&nbsp;</label>
+                    <b-form-input
+                        id="tool-name-textbox"
+                        v-model="newToolGroupName"
+                        :placeholder="uiText.toolGroupNamePlaceholder" />
+                </b-input-group>
+            </b-row>
+
+            <!-- Columns in tool group -->
+            <b-row>
+                <b-input-group>
+                    <label for="column-multiselect">Columns:&nbsp;</label>
+                    <b-form-select
+                        id="column-multiselect"
+                        v-model="selectedTools"
+                        multiple
+                        :options="columnOptions()"
+                        :select-size="4" />
+                </b-input-group>
+            </b-row>
+
+            <!-- Create new tool group button -->
+            <b-row>
+                <b-col cols="4">
+                    <!-- Error message - only visible if duplicate group name is entered -->
+                    <p v-if="invalidNameEntered" class="instructions-text">
+                        {{ uiText.alreadyExistsText }}
+                    </p>
+                </b-col>
+                <b-col class="no-padding-right" ols="8">
+                    <b-button
+                        class="float-right"
+                        :disabled="!readyToAddToolGroup"
+                        variant="info"
+                        @click="createToolGroup()">
+                        {{ uiText.createToolGroupButton }}
+                    </b-button>
+                </b-col>
+            </b-row>
+
+        </b-col>
+
+        <b-col cols="6">
+
+            <!-- Tool grouping table -->
+            <b-table
+                bordered
+                head-variant="dark"
+                :items="assessmentToolGroups.items"
+                :fields="assessmentToolGroups.fields">
+
+                <!-- Blank row for empty table for aesthetics -->
+                <template #top-row v-if="hasNoGroups">
+                    <!-- Adding &nbsp; to the cell so that it maintains the standard cell height -->
+                    <td
+                        v-for="index in assessmentToolGroups.fields.length"
+                        :key="assessmentToolGroups.fields[index]">
```

Comment:
Did you mean to add `key` here? I get a bunch of warnings about non-literal keys and duplicate keys:

![image](https://user-images.githubusercontent.com/1302022/167905978-1560a230-0254-4a5f-9a14-dc60971cdd3c.png)

Reply:
To clarify, I mean: `assessmentToolGroups.fields[index].key`?

Reply:
I caught this one in my current work. Didn't see the warnings. The fix is

```
                        v-for="field in assessmentToolGroups.fields"
                        :key="field.key">
```